pkg/distributor/forwarding/forwarding.go

```diff
@@ -253,7 +262,25 @@ type tsByTargets map[string]tsWithSampleCount
 
 // copyToTarget copies the given time series into the given target and does the necessary accounting.
 // The time series is deep-copied, so the passed in time series can be returned to the pool without affecting the copy.
-func (t tsByTargets) copyToTarget(target string, ts mimirpb.PreallocTimeseries, pool *pools) {
+func (t tsByTargets) copyToTarget(target string, dontForwardBeforeTs int64, ts mimirpb.PreallocTimeseries, pool *pools) (err error) {
+	if dontForwardBeforeTs > 0 && samplesNeedFiltering(ts.TimeSeries.Samples, dontForwardBeforeTs) {
+		samplesUnfiltered := ts.TimeSeries.Samples
+		defer func() {
+			ts.TimeSeries.Samples = samplesUnfiltered
```

Comment:
On line 281 we set `ts.TimeSeries.Samples = samplesFiltered`. This defer function then undoes that? Why?

Reply:
because if the `ingest` flag is set to `true` then we still want to ingest the unfiltered samples.

Reply:
added a comment ae9d2006d450a625c827cd42f8b98e1085af97a1

Reply:
Thanks, now I understand what's happening. I added one more comment about preparing `ts.TimeSeries` for deep-copy.